### ISSUE TITLE
error message not in perldiag

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2287]
Migrated from [rt.perl.org#3606](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3606) (status was 'resolved')

Searchable as RT3606$

**Comment by p5pRT** [comment]
### From bgriffin@cddb.com

#### Created by bgriffin@cddb.com




While checking with \-wc a script that had a syntax error I
got an odd error message not described in perldiag\.

$ perl5\.6\.0 \-wc
use strict;
use vars qw\( $a $b $c $syntaxerror;
&nbsp;          $d $e $f \);
package 'vars' not registered for warnings at \- line 2
BEGIN failed\-\-compilation aborted at \- line 3\.


Incidentally\, perl5\.00503 did not highlight this error\, but
reported an 'Global symbol "$syntaxerror" requires explicit
package name' error when I first used the variable\, which
got me to find the typo\.


Unrelated\, but I notice in the config information below that
the archname is misleading\. I originally tried to build a
multiplicity enabled perl\, but the compile failed so I did
a make clean and reran configure\, only changing that one
option\, but the archname did not change\. Is that a bug or
an improper build?

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=medium

Site configuration information for perl v5.6.0:

Configured by bgriffin at Fri Jul 28 14:55:44 PDT 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.2.5-15, archname=i686-linux-thread-multi
    uname='linux bgriffin-unx 2.2.5-15 #1 mon apr 19 23:00:46 edt 1999 i686 unknown '
    config_args=''
    hint=previous, useposix=true, d_sigaction=define
    usethreads=define use5005threads=define useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=define usesocks=undef
  Compiler:
    cc='cc', optimize='-O2', gccversion=egcs-2.91.66 19990314/Linux (egcs-1.1.2 release)
    cppflags='-D_REENTRANT -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    ccflags ='-D_REENTRANT -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='long double', nvsize=12, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldb -ldl -lm -lpthread -lc -lposix -lcrypt
    libc=/lib/libc-2.1.1.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /home/bgriffin/perl5lib
    /usr/lib/perl5/5.6.0/i686-linux-thread-multi
    /usr/lib/perl5/5.6.0
    /usr/lib/perl5/site_perl/5.6.0/i686-linux-thread-multi
    /usr/lib/perl5/site_perl/5.6.0
    /usr/lib/perl5/site_perl/5.005
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/bgriffin
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/bgriffin/bin:/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin:/sbin:/usr/bin:/usr/sbin
    PERL5LIB=/home/bgriffin/perl5lib
    PERL_BADLANG (unset)
    SHELL=/bin/ksh


```
</details>

**Comment by p5pRT** [comment]
### From The RT System itself

fixed in change \#7408
